modules/core/04-channel/types/upgrade.go

```diff
@@ -77,12 +77,48 @@ func (ut Timeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
 
+// UpgradeError defines an error that occurs during an upgrade.
+type UpgradeError struct {
+	// underlyingError is the underlying error that caused the upgrade to fail.
+	// this error should not be written to state.
+	underlyingError error
```

Comment:
what do you think about `err`, `cause` or `reason` as possible alternatives for this field?

Reply:
yep absolutely, I think this name was mostly to show intent in the github issue! happy to just have this as even just `err`